R/sun-methods.R

```diff
@@ -170,20 +247,36 @@ setMethod("solarnoon", signature(crds="matrix", dateTime="POSIXct"),
                         POSIXct.out=POSIXct.out)
           })
 
+#' @description Method for SpatialPoints objects.
+#' @rdname solarnoon
 setMethod("solarnoon", signature(crds="SpatialPoints", dateTime="POSIXct"),
           function(crds, dateTime, POSIXct.out=FALSE) {
               crds.sf <- st_as_sf(crds)
               solarnoon(crds.sf, dateTime=dateTime,
                         POSIXct.out=POSIXct.out)
           })
 
-###_ + solarpos methods
-if (!isGeneric("solarpos")) {
-    setGeneric("solarpos", function(crds, dateTime, ...) {
-        standardGeneric("solarpos")
-    })
-}
+#' @title Compute solar position
+#' @description Calculates the solar position, i.e., the sun's elevation and azimuth, at a specific geographical location and time. 
+#' Methods are available for different object types with geographical coordinates, including:
+#' * "sf": an object of class "sf".
+#' * "matrix": a matrix of coordinates.
+#' * "SpatialPoints": an object of class "SpatialPoints".
+#'
+#' @param crds This represents geographical coordinates. It can be an object of
```

Comment:
many of these arguments are the same as for the other functions. You can reduce the documentation and make maintenance easier by using `@inheritParams sunriset`, which will recycle the documentation in sunriset for any argument not explicitly specified.